chapters/4/konteneryzacja.tex

```diff
@@ -0,0 +1,42 @@
+
+\subsubsection{Silnik konteneryzacji}
+Konteneryzacja to technologia, pozwalająca na uzyskanie większości funkcjonalności wirtualizacji wykorzystując stosunkowe małe maszyny logiczne nazywane "kontenerami". Główna różnica pomiędzy wirtualizacją i konteneryzacją polega na wykorzystaniu systemu operacyjnego gospodarza. O ile w przypadku wirtualizacji, z wykorzystaniem nadzorcy typu pierwszego i drugiego, każda maszyna wirtualna posiada swój własny kernel (jądro) systemu operacyjnego, w przypadku konteneryzacji system operacyjny, w szczególności jego jądro, jest współdzielone między wszystkie kontenery. Skutkuje to zmniejszeniem rozmiaru kontenerów, które nie muszą zawierać w sobie obrazu całego systemu i szybszym czasem uruchamiania, które są często głównymi czynnikami przemawiającymi za zastosowaniem kontenerów nad maszynami wirtualnymi. Jednakże, z racji ścisłego powiązania z systemem gospodarza, na którym działa kontener, lemat dotyczący zachowania relacji równoważności instrukcji dotyczący maszyny wirtualnej Popka i Goldberga \cite{virtualization} nie jest zachowany dla każdego kontenera uruchomionego na tej samej architekturze. Przykładowo, kontener oparty na systemie Linux może zostać uruchomiony na maszynach gospodarza o różnych wersjach kernela (jądra) systemu, rezultatem czego może być różna interpretacja wywołań systemowych dla identycznych wywołań w każdym z tych kontenerów. Dlatego, w świetle tego artykułu, konteneryzacja nie jest stricte rodzajem wirtualizacji. Z praktycznego punktu widzenia, oznacza to niemożność wykorzystania kontenerów opartych na pewnym systemie operacyjnym $O_a$ na maszynie o systemie operacyjnym $O_b$, gdzie kernele (jądra) systemów $O_a$ i $O_b$ są ze sobą niekompatybilne -- przykładem takiej sytuacji jest próba uruchomienia kontenera korzystającego z systemu Microsoft Windows ($O_a$) na maszynie opartej na systemie Linux ($O_b$).
+
+\subsection{Wirtualizacja a platforma serwerowa dla systemu STOS}
+Z racji wystąpienia komplikacji, o których była mowa w poprzednim podrozdziale, w ramach poszukiwania rozwiązania alternatywnego dla konteneryzacji, naturalnym wyborem była analiza wirtualizacji jako technologii stanowiącej podstawę wdrożenia systemu. Pierwszą decyzją, którą należało podjąć, był wybór typu nadzorcy \textit{(hypervisor)} i konkretnej implementacji tego rozwiązania. W świetle doświadczeń z systemami orkiestracji kontenerów naturalnym wyborem był system oparty na nadzorcy typu pierwszego \textit{(type 1 hypervisor)}, wiele implementacji którego oferuje funkcjonalności z zakresu zarządzania wieloma maszynami fizycznymi, co było jednym z wymagań dla wdrożonego systemu. Dwiema implementacjami takiego systemu, których zastosowanie było rozważane były Proxmox i xcp-ng \cite{proxmox, xcp}. Oba z tych rozwiązań oferują bardzo podobne rozwiązania, lecz ostatecznie wybrano xcp-ng jako system nadzorcy, na podstawie którego realizowana jest wirtualizacja serwisów w ramach platformy serwerowej dla systemu STOS.
```

Comment:
> wiele implementacji którego oferuje funkcjonalności z zakresu

To dziwnie brzmi

> których zastosowanie było rozważane były Proxmox

których zastosowanie było rozważane, były Proxmox (przecinek)